### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'timeoutType')

### DISCUSSION THREAD

**Discussion by magnusrodseth** [nodejs/undici#2035] [Q&A]
To set the stage, I am using a Turborepo with Next.js, tRPC and Clerk for authentication and user management. 

As a preface: I started by asking in the Discussion in the [tRPC repository](https://github.com/trpc/trpc/discussions/4112#discussioncomment-5474697), and the answer I got there was the following:

> It's Node.js, and to be more specific, it's Unidici, their HTTP client. There are only 2 places where it can break there - [nodejs/undici@dc8d111/lib/client.js#L1207-L1220](https://github.com/nodejs/undici/blob/dc8d11183fb95aa93d296bde3e21501856849367/lib/client.js#L1207-L1220)

**Hence, I'm asking here, so here goes.**

Ever since the start of my project, I sometimes get this error message:

```sh
@acme/web:dev: error - unhandledRejection: TypeError: Cannot read properties of undefined (reading 'timeoutType')
@acme/web:dev:     at _resume (evalmachine.<anonymous>:11000:33)
@acme/web:dev:     at resume (evalmachine.<anonymous>:10969:7)
@acme/web:dev:     at connect (evalmachine.<anonymous>:10956:7)
@acme/web:dev: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
@acme/web:dev: TypeError: Cannot read properties of undefined (reading 'timeoutType')
@acme/web:dev:     at _resume (evalmachine.<anonymous>:11000:33)
@acme/web:dev:     at resume (evalmachine.<anonymous>:10969:7)
@acme/web:dev:     at connect (evalmachine.<anonymous>:10956:7)
```

The error happens sporadically, and I have not been able to pinpoint any pattern of it occurring. The thing that baffles me is that when I Google this error, no other discussions around it exist.

It may happen when fetching data asynchronously between the tRPC client and server.

The stack trace is also very challenging to navigate, and does not offer much help. I have now concluded that `timeoutType` is something related to `undici`.

**Do you have any general recommendations on how to debug and fix this issue? Do you have an idea under what circumstances this can be triggered?**

**Comment by ronag** [comment]
As far as I know it happens on older versions of undici when it is unable to load the wasm binaries,e.g. when there isn't sufficient memory.

**Reply by magnusrodseth** [reply]
@ronag How can I check my `undici` version? I have not installed it manually, so it must be a peer dependency of some library?

**Reply by ronag** [reply]
Can't help you with that.
